### PR TITLE
browser(firefox): preserve content-type from original request if it i…

### PR DESCRIPTION
**Pull request by yury-s**
…s not overridden

Reference #16736

### CODE REVIEW

File: browser_patches/firefox-beta/juggler/NetworkObserver.js [hunk 1]
```diff
@@ -863,19 +863,30 @@ function setPostData(httpChannel, postData, headers) {
...
-    return defaultValue;
+    return undefined;
   }
   // Clear content-length, so that upload stream resets it.
   httpChannel.setRequestHeader('content-length', '', false /* merge */);
-  httpChannel.explicitSetUploadStream(synthesized, overriddenHeader('content-type', 'application/octet-stream'), -1, httpChannel.requestMethod, false);
+  let contentType = overriddenHeader('content-type');
+  if (contentType === undefined) {
+    try {
+      contentType = httpChannel.getRequestHeader('content-encoding');
```
Comment (rwoll) [inline browser_patches/firefox-beta/juggler/NetworkObserver.js:881]: Is this correct? What happens with gzipped req/resp? Would we report content-type as `gzip` b/c of this line? That doesn't sound correct.

Comment (yury-s) [reply browser_patches/firefox-beta/juggler/NetworkObserver.js:881]: Yes this is the intended behavior unless content-type is explicitly overridden we use the one from the original request. That's what we do in other browsers too.

Comment (yury-s) [reply browser_patches/firefox-beta/juggler/NetworkObserver.js:881]: Do you suggest something different?

Comment (yury-s) [reply browser_patches/firefox-beta/juggler/NetworkObserver.js]: Oh, you meant the typo 'content-encoding' instead of 'content-type', fixed.

**Review by dgozman — APPROVED** [review]

**Pull request merged**